specs/_features/eip7805/validator.md

```diff
@@ -85,39 +123,89 @@ def get_inclusion_committee_assignment(
 
-Some validators are selected to submit signed inclusion list. Validators should
-call `get_inclusion_committee_assignment` at the beginning of an epoch to be
-prepared to submit their inclusion list during the next epoch.
+*Note*: The proposer should call `prepare_execution_payload` at
+`PROPOSER_INCLUSION_LIST_CUT_OFF` into the slot with the
+`inclusion_list_transactions` field of `PayloadAttributes` set to the inclusion
+list transactions gathered up to `PROPOSER_INCLUSION_LIST_CUT_OFF` into the slot
+to ensure the execution payload satisfies the inclusion list constraints.
 
-A validator should create and broadcast the `signed_inclusion_list` to the
-global `inclusion_list` subnet by `PROPOSER_INCLUSION_LIST_CUT_OFF` seconds into
-the slot, unless a block for the current slot has been processed and is the head
-of the chain and broadcast to the network.
+```python
+def prepare_execution_payload(
+    state: BeaconState,
+    safe_block_hash: Hash32,
+    finalized_block_hash: Hash32,
+    suggested_fee_recipient: ExecutionAddress,
+    execution_engine: ExecutionEngine,
+) -> Optional[PayloadId]:
+    # Verify consistency of the parent hash with respect to the previous execution payload header
+    parent_hash = state.latest_execution_payload_header.block_hash
+
+    # Set the forkchoice head and initiate the payload build process
+    withdrawals, _ = get_expected_withdrawals(state)
+
+    # [New in EIP7805]
+    # Get the inclusion list store
+    inclusion_list_store = get_inclusion_list_store()
+
+    payload_attributes = PayloadAttributes(
+        timestamp=compute_time_at_slot(state, state.slot),
+        prev_randao=get_randao_mix(state, get_current_epoch(state)),
+        suggested_fee_recipient=suggested_fee_recipient,
+        withdrawals=withdrawals,
+        parent_beacon_block_root=hash_tree_root(state.latest_block_header),
+        inclusion_list_transactions=get_inclusion_list_transactions(
+            inclusion_list_store, state, Slot(state.slot - 1)
+        ),  # [New in EIP7805]
+    )
+    return execution_engine.notify_forkchoice_updated(
+        head_block_hash=parent_hash,
+        safe_block_hash=safe_block_hash,
+        finalized_block_hash=finalized_block_hash,
+        payload_attributes=payload_attributes,
+    )
+```
 
-#### Constructing a signed inclusion list
+### Inclusion list proposal
+
+A validator is expected to propose a
+[`SignedInclusionList`](./beacon-chain.md#signedinclusionlist) at the beginning
+of any `slot` for which
+`get_inclusion_committee_assignment(state, epoch, validator_index)` returns.
+
+If a validator is in the current inclusion list committee, the validator should
+create and broadcast the `signed_inclusion_list` to the global `inclusion_list`
+subnet by `INCLUSION_LIST_SUBMISSION_DEADLINE` seconds into the slot after
+processing the block for the current slot and confirming it as the head. If no
+block is received by `INCLUSION_LIST_SUBMISSION_DEADLINE - 1` seconds into the
```

Comment:
I think there might need to be a config variable for this too. Let's deal with this later though.

Reply:
I think I added it at L54. Do you mean preset?

Reply:
Sorry, this was ambiguous. I'm referring to `INCLUSION_LIST_SUBMISSION_DEADLINE - 1`.

I'm saying that we might want a new `INCLUSION_LIST_BLOCK_DEADLINE` or something.

Let's deal with this later. Not sure if this is a good idea or not.